src/Sulu/Component/Content/Document/Behavior/RouteBehavior.php

```diff
@@ -31,4 +31,32 @@ public function getTargetDocument();
      * @param object $targetDocument
      */
     public function setTargetDocument($targetDocument);
+
+    /**
+     * Returns locale.
+     *
+     * @return string
+     */
+    public function getLocale();
+
+    /**
+     * Sets locale.
+     *
+     * @param string $locale
+     */
+    public function setLocale($locale);
+
+    /**
+     * Returns history flag.
+     *
+     * @return bool
+     */
+    public function isHistory();
+
+    /**
+     * Set history flag.
+     *
+     * @param bool $history
+     */
+    public function setHistory($history);
```

Comment:
this would be quite hard because then we have to extract the functionality for creating, updating and removing history. otherwise you suggest that a user can create a route with history but it doesn't work because the logic is bound to something else.